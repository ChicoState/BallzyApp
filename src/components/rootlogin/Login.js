'use strict';
import {
  AppRegistry,
  AsyncStorage,
  View,
  ToolbarAndroid,
  ActivityIndicator,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Text,
} from 'react-native';

import React, {Component} from 'react';
import styles from '../../styles/LoginStyles';
import firebaseApp from '../../globals'

export default class Login extends Component {
  static navigationOptions = {
    header: null,
  }
  constructor(props) {
    super(props);
    this.title = "Login:";
    this.state = {
      email: '',
      password: '',
      loading: false
    }
  }
  login() {
    this.setState({
      loading: true
    });
    firebaseApp.auth().signInWithEmailAndPassword(this.state.email, this.state.password
    ).then((userData) => {
        this.setState({
          loading: false
        });
        AsyncStorage.setItem('userData', JSON.stringify(userData));
        this.props.navigation.navigate("Mainapp");
      }
    ).catch((error) => {
      this.setState({
        loading: false
      });
      alert('Login Failed. Please try again'+error);
      });
  }
  render() {
    if (this.state.loading == false) {
      return(
        <View style={styles.container}>
          <View style={styles.header}>
            <View style={{flex:2}}/>
            <View style={{flex:6}}>
              <View style={styles.appnamecontainer}/>
              <Text style={StyleSheet.flatten(styles.titletext)}>{this.title}</Text>
            </View>
            <View style={{flex:2}}/>
          </View>
          <View style={styles.body}>
            <View style={{flex:2}}/>
            <View style={{flex: 6, flexDirection: 'column'}}>
              <View style={{flex:3, flexDirection: 'column',
                            justifyContent: 'center'}}>
                <TextInput
                  style={styles.btextinput}
                  onChangeText={(text) => this.setState({email: text})}
                  value={this.state.email}
                  placeholder={"Email Address"}
                />
                <View style={{flex: .5}}/>
                <TextInput
                  style={styles.btextinput}
                  onChangeText={(text) => this.setState({password: text})}
                  value={this.state.password}
                  secureTextEntry={true}
                  placeholder={"Password"}
                />
                <View style={{flex: 9.5, flexDirection: 'column',
                              justifyContent: 'center'}}>

                  <TouchableOpacity onPress={() => this.login()}>
                    <View style={styles.button}>
                      <Text style={StyleSheet.flatten(styles.buttontext)}>Login</Text>
                    </View>
                  </TouchableOpacity>
                  <TouchableOpacity onPress={() => this.props.navigation.navigate('Signup')}>
                    <View style={styles.transparentbutton}>
                      <Text style={StyleSheet.flatten(styles.buttontexttransparent)}>Need Account?</Text>
                    </View>
                  </TouchableOpacity >
                </View>
              </View>
            </View>
            <View style={{flex:2}}/>
          </View>
        </View>
      )
    } else {
      return(
        <ActivityIndicator size="large"/>
      )
    }
  }

}

AppRegistry.registerComponent('Login', () => Login);
