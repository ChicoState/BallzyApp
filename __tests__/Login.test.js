'use strict';

import 'react-native';
import React from 'react';
//import React from 'react';
import Login from '../src/components/rootlogin/Login';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Login />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
