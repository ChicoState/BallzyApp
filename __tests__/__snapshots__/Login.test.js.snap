exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 3,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }>
    <View
      style={
        Object {
          "flex": 2,
        }
      } />
    <View
      style={
        Object {
          "flex": 6,
        }
      }>
      <View
        style={
          Object {
            "flex": 6,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#0C2C52",
            "fontSize": 30,
          }
        }>
        Login:
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 2,
        }
      } />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 7,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }>
    <View
      style={
        Object {
          "flex": 2,
        }
      } />
    <View
      style={
        Object {
          "flex": 6,
          "flexDirection": "column",
        }
      }>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }>
        <TextInput
          onChangeText={[Function]}
          placeholder="Email Address"
          style={
            Object {
              "backgroundColor": "#DCF0F7",
              "borderWidth": 0,
              "height": 40,
            }
          }
          value="" />
        <View
          style={
            Object {
              "flex": 0.5,
            }
          } />
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#DCF0F7",
              "borderWidth": 0,
              "height": 40,
            }
          }
          value="" />
        <View
          style={
            Object {
              "flex": 9.5,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#5E9DC8",
                  "height": 45,
                  "justifyContent": "center",
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }>
                Login
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "height": 45,
                  "justifyContent": "center",
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#5F6B61",
                    "fontSize": 14,
                  }
                }>
                Need Account?
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 2,
        }
      } />
  </View>
</View>
`;
